### PR TITLE
feat: utils package

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#837
Moves general stateless functions into a utils package.
Also added documentation for Crypto utils and added tests as the coverage was too low.

N.B.: I simplified code on most decoder functions as we had a check on there that has become redundant.

## How to test:
Tests, as usual

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by rflechtner** [comment]
Seems like we need to figure out where to put development tools that should not be published, but are used during development (mocking tools, jest add-ons). This could either be a directory somewhere or a package which we don't publish and import as a dev dependency.

**Comment by rflechtner** [comment]
Albi broke the sdk integration again but it tests successfully against yesterday's dev node.

**Comment by weichweich** [comment]
> Albi broke the sdk integration again but it tests successfully against yesterday's dev node.

just making sure that my PR gets merged first! 😝

**Comment by rflechtner** [comment]
> Albi broke the sdk integration again but it tests successfully against yesterday's dev node.

that's fixed !

**Comment by weichweich** [comment]
> > Albi broke the sdk integration again but it tests successfully against yesterday's dev node.
> 
> that's fixed !

and it wasn't even my fault! 😮

File: packages/utils/tsconfig.build.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+   "include": [
+    "src/**/*.ts", "src/**/*.js"
+  ],
+
+   "exclude": [
+    "coverage",
+    "**/*.spec.ts",
+  ]
+}
```
Comment (LeonFLK) [inline packages/utils/tsconfig.build.json:16]:
```suggestion
}

```

EOF newline

**Review by LeonFLK — COMMENTED** [review]
LGTM

Nitpick: add EOF newline.
Other than that, do you think the `SubscriptionPromise.ts` should live inside the utils package as well?

**Comment by rflechtner** [comment]
> LGTM
> 
> Nitpick: add EOF newline.
> Other than that, do you think the `SubscriptionPromise.ts` should live inside the utils package as well?

Timo & I thought it makes more sense on the BlockchainUtils. But it is something we can consider, it's still not clear to me where it makes most sense

**Comment by LeonFLK** [comment]
> > LGTM
> > Nitpick: add EOF newline.
> > Other than that, do you think the `SubscriptionPromise.ts` should live inside the utils package as well?
> 
> Timo & I thought it makes more sense on the BlockchainUtils. But it is something we can consider, it's still not clear to me where it makes most sense

Yes that's what I thought, either inside utils or blockchain helper/utils.

**Review by LeonFLK — APPROVED** [review]
LGTM



PromiseSubscription goes into Blockchain Helpers

**Pull request merged**